### PR TITLE
WIP BXMSDOC-5984: Added best practice for Kie server endpoints in DMN

### PR DESCRIPTION
**Pull request by hmanwani-rh**
See JIRA: https://issues.redhat.com/browse/BXMSDOC-5984

Added the best practice for the REST endpoint of KIE Server.

Doc preview
RHPAM - [Designing a decision service using DMN models](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-5984-RHPAM/#dmn-execution-rest-proc_dmn-models)
RHDM - [Designing a decision service using DMN models](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-5984-RHDM/#dmn-execution-rest-proc_dmn-models)

### CODE REVIEW

File: doc-content/drools-docs/src/main/asciidoc/DMN/dmn-execution-rest-proc.adoc [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 Directly interacting with the REST endpoints of {KIE_SERVER} provides the most separation between the calling code and the decision logic definition. The calling code is completely free of direct dependencies, and you can implement it in an entirely different development platform such as `Node.js` or `.NET`. The examples in this section demonstrate Nix-style curl commands but provide relevant information to adapt to any REST client.
 
+When you use a REST endpoint of KIE Server, the best practice is to define a domain object POJO Java class, annotated with standard KIE Server marshalling annotations.
```
Comment (tarilabs) [inline doc-content/drools-docs/src/main/asciidoc/DMN/dmn-execution-rest-proc.adoc]:
```suggestion
When you use a REST endpoint of {KIE_SERVER}, the best practice is to define a domain object POJO Java class, annotated with standard {KIE_SERVER} marshalling annotations.
```

**Review by tarilabs — COMMENTED** [review]
how about using the expected variable? Or was this done on purpose?

Just a thought, otherwise is fine, thank you!

File: doc-content/drools-docs/src/main/asciidoc/DMN/dmn-execution-rest-proc.adoc (hunk 1 above)
Comment (hmanwani-rh) [reply doc-content/drools-docs/src/main/asciidoc/DMN/dmn-execution-rest-proc.adoc]: Oops, thanks for catching that Matteo.

**Review by kaldesai — APPROVED** [review]
Hey @hmanwani-rh, Your changes look good to me. Thank you.

**Review by jomarko — APPROVED** [review]

**Pull request merged**